### PR TITLE
improve data download feature

### PR DESCRIPTION
**Pull request by smcalilly**
## Overview

For #816 and #807

Includes:
- additional make recipes for filtering data, based on the list in [#816](https://github.com/security-force-monitor/sfm-cms/issues/816#issuecomment-1269865342) comment
- updated the download template with text in [#816](https://github.com/security-force-monitor/sfm-cms/issues/816#issuecomment-1269865342) comment
- removed the old download buttons
- added nav link to download page
- blank readme for demo (per tom's instruction in an email)
- new import docket with country name
- updated the download data import to use country name

### Notes
The data archive is a flat directory. The recipes write to `<country_name>_<entity_type>.csv` instead of creating a new directory with the country name. I couldn't figure out how to cleanly create a new directory based on the dynamic country name values. Let me know if you see a way to do this?

I had to create a custom processing script to blank out the values for two specific columns based on a discussion in our slack #computer-programming channel. For context: csvgrep can only remove the columns. csvsql can change the column values, but csvsql would completely remove the header row from a file that had no data rows, thus creating a blank file which isn't something we want.

## Testing Instructions
- get aws s3 token from `wwic-data-archive-staging token` in last pass
- `cp .env.s3.example .env` and add your AWS access tokens
- for testing purposes, you might need to remove the myanmar row from the import docket, since that spreadsheet naming convention will break the import (this issue has been mentioned in an email, to be fixed)
- to create an archive locally, run `docker-compose --env-file .env.s3 run --rm app make data_archive`
- visit `localhost:8000/en/download/` to download the data

### CODE REVIEW

File: data/processors/blank_columns.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+# init arg parser
+parser = argparse.ArgumentParser()
+parser.add_argument('--entity', type=str, required=True)
+args = parser.parse_args()
+
+# init the incoming data as a dict reader
+lines = iter(line.decode('utf-8').strip() for line in sys.stdin.buffer.readlines())
+header = next(lines).split(',')
+reader = csv.DictReader(lines, fieldnames=header)
```
Comment (hancush) [inline data/processors/blank_columns.py:13]: Hm, can you not do:

```python
reader = csv.DictReader(sys.stdin)
```

[If fieldnames is omitted, the values in the first row of file f will be used as the fieldnames.](https://docs.python.org/3/library/csv.html#csv.DictReader)

File: data/processors/blank_columns.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+# write data to stdout
+stdout_csv = csv.writer(sys.stdout)
+stdout_csv.writerow(header)
```
Comment (hancush) [inline data/processors/blank_columns.py:17]: I would do:

```
stdout_csv = csv.DictWriter(sys.stdout, fieldnames=reader.fieldnames)
```

https://docs.python.org/3/library/csv.html#reader-objects

File: data/processors/blank_columns.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+for row in reader:
+    row.update({
+        f'{args.entity}:comments:admin': '',
+        f'{args.entity}:owner:admin': ''
+    })
+    stdout_csv.writerow(row.values())
```
Comment (hancush) [inline data/processors/blank_columns.py:24]: And then you could just do `writerow(row)`

File: docket.mk [hunk 4]
```diff
@@ -1,35 +1,68 @@
 .PHONY: sfm_pc/management/commands/country_data
 
-%_import : %.csv sfm_pc/management/commands/country_data
+DATA_ARCHIVE_BUCKET := $(shell cat configs/s3_config.json | jq -r '.data_archive_bucket')
```
Comment (hancush) [inline docket.mk:3]: Always impressed when I see you use jq.

File: docket.mk [hunk 5]
```diff
@@ -1,35 +1,68 @@
...
+ENTITIES=units.csv persons.csv incidents.csv locations.csv locations.geojson
+
+.PHONY : filtered_data
+filtered_data: $(foreach country,$(COUNTRY_NAMES),$(patsubst %,data/wwic_download/$(country)_%,$(ENTITIES)))
+	echo "filtered csvs for entities"
+
+define filter_entity_data
+	$(shell csvgrep --columns $(1):status:admin --match 3 $< | \
+					python data/processors/blank_columns.py --entity $(1) > $@)
+endef
```
Comment (hancush) [inline docket.mk:22]: Nice!

**Review by hancush — CHANGES_REQUESTED** [review]
Hey, @smcalilly, great refactor! Once you respond to my super minor comments, let's get this up on staging for review.

File: data/processors/blank_columns.py (hunk 1 above)
Comment (smcalilly) [reply data/processors/blank_columns.py]: nice! much better.

**Comment by smcalilly** [comment]
@hancush I made that python change, thanks for that.

When I added the individual sources file to the pipeline, it was sometimes missing a column and erroring the `blank_columns.py` script. So I did some explicit checks for that: https://github.com/security-force-monitor/sfm-cms/blob/5a944f65afe6a1a8ac464987ae808a8d46d322d4/data/processors/blank_columns.py#L17

Not the cleanest but it works.

**Review by hancush — APPROVED** [review]

**Pull request merged**